runtime/j9vm/javanextvmi.cpp

```diff
@@ -272,8 +273,24 @@ exitVThreadTransitionCritical(J9VMThread *currentThread, j9object_t vthread)
 	J9OBJECT_I64_STORE(currentThread, vthread, currentThread->javaVM->virtualThreadInspectorCountOffset, 0);
 }
 
-JNIEXPORT void JNICALL
-JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
+#if JAVA_SPEC_VERSION >= 20
+static void
+virtualThreadHideFrames(J9VMThread *currentThread, j9object_t vthread, BOOLEAN hide) {
+	Assert_SC_true(IS_JAVA_LANG_VIRTUALTHREAD(currentThread, currentThread->threadObject));
```

Comment:
`J9VMThread->threadObject` will change between `virtualThread[M|Un]ountBegin` and `virtualThread[M|Un]ountEnd`. `J9VMThread->threadObject` won't always be a virtual thread.

Reply:
Hide frames should always be called from a virtual thread object to notify VM of modification to its currents states which should be ignored. this behaviour doesn't affect the mount/unmount operation as `HideFrames` should only be called by mounted virtual thread.

Reply:
This assertion is triggered when the `jvmti/events/FramePop/framepop02` test (https://github.com/eclipse-openj9/openj9/issues/16346) is run with the changes from this PR. Without the assertion, the failing `framepop02` test passes with the changes from this PR. The logic needs a revision to avoid triggering this assertion.

Reply:
Ok, i see what you mean, the assertion should be placed directly inside `JVM_notifyJvmtiHideFrames` instead of the helper.

Reply:
@thallium can yo move the assertion such that only `JVM_VirtualThreadHideFrames` triggers that.